repos/system_upgrade/el7toel8/actors/vsftpdconfigread/libraries/library.py

```diff
@@ -0,0 +1,51 @@
+import errno
+import os
+
+from leapp.libraries.actor import config_parser
+from leapp.libraries.common.vsftpdutils import get_config_contents, get_default_config_hash, \
+                                               read_file, \
+                                               VSFTPD_CONFIG_DIR, \
+                                               STRICT_SSL_READ_EOF, TCP_WRAPPERS
+from leapp.libraries.stdlib import api
+from leapp.models import VsftpdConfig, VsftpdFacts
+
+
+def _parse_config(path, content):
+    try:
+        parser = config_parser.VsftpdConfigParser(content)
+        return parser.parsed_config
+    except config_parser.ParsingError:
+        api.current_logger().info('File %s does not look like vsftpd configuration, skipping.'
+                                  % path)
+        return None
+
+
+def _get_parsed_configs(read_func=read_file, listdir=os.listdir):
+    res = []
+    try:
+        for fname in listdir(VSFTPD_CONFIG_DIR):
+            path = os.path.join(VSFTPD_CONFIG_DIR, fname)
+            if not path.endswith('.conf'):
+                continue
+            content = get_config_contents(path, read_func=read_func)
+            if content is None:
+                continue
+            parsed = _parse_config(path, content)
+            if parsed is not None:
+                res.append((path, parsed))
+    except IOError as e:
```

Comment:
OSError: [Errno 2] No such file or directory: '/etc/vsftpd'

```suggestion
    except OSError as e:
```

Reply:
Thanks again. Hopefully fixed now. I also noticed that I wasn't creating IOError and OSError in the tests correctly (the errno attribute wasn't being set), so I fixed that. And rebased against origin/master.

Reply:
Notice that there's a newly added first commit.